### PR TITLE
Add support for getting isDefaultVersion details for an API in Internal data service

### PR DESCRIPTION
**Pull request by tharikaGitHub**
This PR adds the functionality to support getting **isDefaultVersion** details for an API in Internal data service.

Related PR : https://github.com/wso2/carbon-apimgt/pull/9044

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SubscriptionValidationSQLConstants.java [hunk 1]
```diff
@@ -348,161 +366,256 @@
...
+                    "         AND API.API_ID = PROD.API_ID" +
+                    "         AND API.API_VERSION = ?" +
+                    "         AND API.CONTEXT = ?" +
+                    "   )" +
+                    "   AS APIS " +
+                    "   LEFT JOIN " +
+                    "      AM_API_DEFAULT_VERSION AS DEF " +
+                    "      ON APIS.API_NAME = DEF.API_NAME" +
+                    "      AND APIS.API_PROVIDER = DEF.API_PROVIDER" +
+                    "      AND APIS.API_VERSION = DEF.DEFAULT_API_VERSION";
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SubscriptionValidationSQLConstants.java:618]: @tharikaGitHub don't we hv to get the PUBLISHED_DEFAULT_API_VERSION? shall we check this

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SubscriptionValidationSQLConstants.java [hunk 2]
```diff
@@ -348,161 +366,256 @@
...
+                    "         API.API_ID = URL.API_ID" +
+                    "         AND API.API_VERSION = ?" +
+                    "         AND API.CONTEXT = ?" +
+                    "   )" +
+                    "   AS APIS " +
+                    "   LEFT JOIN " +
+                    "      AM_API_DEFAULT_VERSION AS DEF" +
+                    "      ON APIS.API_NAME = DEF.API_NAME" +
+                    "      AND APIS.API_PROVIDER = DEF.API_PROVIDER" +
+                    "      AND APIS.API_VERSION = DEF.DEFAULT_API_VERSION";
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SubscriptionValidationSQLConstants.java:565]: @tharikaGitHub don't we hv to get the PUBLISHED_DEFAULT_API_VERSION? shall we check this

Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SubscriptionValidationSQLConstants.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/9076/commits/f19e30fc575d02246c5ba429b9de8aabf8551162

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java [hunk 3]
```diff
@@ -329,6 +329,7 @@ public API getApi(String version, String context) {
                     api.setPolicy(resultSet.getString("API_TIER"));
                     api.setVersion(resultSet.getString("API_VERSION"));
                     api.setContext(resultSet.getString("CONTEXT"));
+                    api.setIsDefaultVersion(resultSet.getBoolean("PUBLISHED_DEFAULT_API_VERSION"));
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java:332]: This logic is not right. "PUBLISHED_DEFAULT_API_VERSION" colum gives you the default version. What you could do is check "PUBLISHED_DEFAULT_API_VERSION" is equal to "API_VERSION" and set the boolean as true if equals

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java [hunk 4]
```diff
@@ -371,6 +372,7 @@ private void populateAPIList(ResultSet resultSet, List<API> apiList) throws SQLE
                 api.setPolicy(resultSet.getString("API_TIER"));
                 api.setVersion(resultSet.getString("API_VERSION"));
                 api.setContext(resultSet.getString("CONTEXT"));
+                api.setIsDefaultVersion(resultSet.getBoolean("PUBLISHED_DEFAULT_API_VERSION"));
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java:375]: same here

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java (hunk 3 above)
Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/9076/commits/9c76d41950a114213dcd434e71f496deec16f689

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java (hunk 4 above)
Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/9076/commits/9c76d41950a114213dcd434e71f496deec16f689

**Pull request merged**